addon/utils/is-promise.js

```diff
@@ -0,0 +1,5 @@
+import RSVP from 'rsvp';
+
+export default function(val) {
+  return val instanceof RSVP.Promise || val instanceof Promise;
```

Comment:
Not sure: wouldn't `val instanceof Promise` throw if `Promise` is undefined (i.e. not supported natively)?